python/ray/tests/test_actor_bounded_threads.py

```diff
@@ -1,92 +1,135 @@
 import sys
 import os
-import psutil
 
 import ray
-import ray.cluster_utils
 import logging
+from typing import Dict
+from collections import Counter
 
 import pytest
 
 logger = logging.getLogger(__name__)
 
+
+def my_threads() -> Dict[str, int]:
+    """
+    Returns [(thread_id, thread_name)]
+    """
+    pid = os.getpid()
+    threads = Counter()
+    proc_dir = f"/proc/{pid}/task"
+
+    for tid_entry in os.listdir(proc_dir):
+        comm_path = os.path.join(proc_dir, tid_entry, "comm")
+
+        if os.path.exists(comm_path):
+            with open(comm_path, "r") as comm_file:
+                thread_name = comm_file.read().strip()
+                threads[thread_name] += 1
+    return threads
+
+
 # Tests a lot of workers sending tasks to an actor, the number of threads for that
 # actor should not infinitely go up.
 
 
-@pytest.mark.skip(reason="Occational extra non-Ray threads, e.g. jemalloc")
+# These therads are from third party code, and may start any time, we can't control
+# them. So we allow them to be any number.
+KNOWN_THREADS = {
+    "grpc_global_tim",  # grpc global timer
+    "grpcpp_sync_ser",  # grpc
+    "jemalloc_bg_thd",  # jemalloc background thread
+}
+
+
+def assert_threads_are_bounded(
+    prev_threads: Dict[str, int], now_threads: Dict[str, int]
+):
+    """
+    Asserts that the threads did not grow unexpected.
+    Rule: For each (thread_name, count) in now_threads, it must either be <= the number
+    in prev_threads, or in KNOWN_THREADS.
+    """
+    for thread_name, count in now_threads.items():
+        if thread_name not in KNOWN_THREADS:
+            target = prev_threads.get(thread_name, 0)
+            assert count <= target, (
+                f"{thread_name} grows unexpectedly: "
+                f"expected <= {target}, got {count}. "
+                f"prev {prev_threads}, now: {now_threads}"
+            )
```

Comment:
what's the current number now? 

Can we have assert outside that calculates num_threads (including known threads) < some number with buffer

to make sure there's no unexpected regressions where known threads are spawned a lot unexpectedly?

Reply:
I tried to play with these 3rd party threads and found they are hard to predict. Sometimes they don't show up and some other times they do. I think it's better to only regulate Ray threads and leave the grpc&jemalloc threads do their thing.